keras_cv/metrics/object_detection/object_detection_metric_suite.py

```diff
@@ -0,0 +1,279 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import sys
+import types
+
+import tensorflow as tf
+import tensorflow.keras as keras
+
+import keras_cv
+from keras_cv import bounding_box
+
+
+class HidePrints:
+    """A basic internal only context manager to hide print statements."""
+
+    def __enter__(self):
+        self._original_stdout = sys.stdout
+        sys.stdout = open(os.devnull, "w")
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        sys.stdout.close()
+        sys.stdout = self._original_stdout
+
+
+def _box_concat(b1, b2):
+    """Concatenates two bounding box batches together."""
+    if b1 is None:
+        return b2
+    if b2 is None:
+        return b1
+
+    result = {}
+    for key in ["boxes", "classes"]:
+        result[key] = tf.concat([b1[key], b2[key]], axis=0)
+
+    if "confidence" in b2:
+        result["confidence"] = tf.concat(
+            [b1["confidence"], b2["confidence"]], axis=0
+        )
+    return result
+
+
+METRIC_NAMES = [
+    "AP",
+    "AP50",
+    "AP75",
+    "APs",
+    "APm",
+    "APl",
+    "ARmax1",
+    "ARmax10",
+    "ARmax100",
+    "ARs",
+    "ARm",
+    "ARl",
+]
+
+
+class ObjectDetectionMetricSuite(keras.metrics.Metric):
+    """ObjectDetectionMetricSuite computes standard object deteciton metrics.
+
+    Args:
+        bounding_box_format: the bounding box format for inputs.
+
+    Usage:
+    `ObjectDetectionMetricSuite()` can be used like any standard metric with any
+    KerasCV object detection model.  Inputs to `y_true` must be KerasCV bounding
+    box dictionaries, `{"classes": classes, "boxes": boxes}`, and `y_pred` must
+    follow the same format with an additional `confidence` key.
+
+    Using this metric suite alongside a model is trivial; simply provide it to
+    the `compile()` arguments of the model:
+
+    ```python
+    images = tf.ones(shape=(1, 512, 512, 3))
+    labels = {
+        "boxes": [
+            [
+                [0, 0, 100, 100],
+                [100, 100, 200, 200],
+                [300, 300, 400, 400],
+            ]
+        ],
+        "classes": [[1, 1, 1]],
+    }
+    model = keras_cv.models.RetinaNet(
+        num_classes=20,
+        bounding_box_format="xywh",
+    )
+
+    # Evaluate model
+    model(images)
+
+    # Train model
+    model.compile(
+        classification_loss='focal',
+        box_loss='smoothl1',
+        optimizer=tf.optimizers.SGD(global_clipnorm=10.0),
+        jit_compile=False,
+        metrics=[keras_cv.metrics.ObjectDetectionMetricSuite('xywh')]
+    )
+    model.fit(images, labels)
+    ```
+    """
+
+    def __init__(self, bounding_box_format, eval_steps, name=None, **kwargs):
```

Comment:
`eval_steps` doesn't really convey the idea of recomputing the output every N step. In Keras "eval steps" kinda sounds like `validation_steps` which is the number of steps to run validation on in `fit()`.

Should probably reuse the arg names used in callbacks that do something every n steps, e.g. `save_freq`. Could be `recompute_freq`.

Reply:
Done - thanks for the input on this API!